rdf-abac-core/src/main/java/io/telicent/jena/abac/core/CxtABAC.java

```diff
@@ -138,6 +138,7 @@ static <X,Y> Cache<X,Y> createHierarchyCache(int size) {
                     Caffeine.newBuilder()
                         .maximumSize(size)
                         .expireAfterWrite(10_000, TimeUnit.MILLISECONDS)
+                        //.recordStats()
```

Comment:
Does it make much of a difference?

Reply:
When the labels are most all the same this cache is not important (per-request label eval caching).

The design issues around the hierarchy cache do not seem to be about cache efficiency so this is there as a note of a possibility. This cache is per request (currently), it could be made system wide but that's outsider the scope of this PR.

The Caffeine javadoc for `recordStats` says it imposes a penalty.

Reply:
Fair points.